app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -793,6 +826,25 @@ public boolean matches(UserKeyRecord ukr) {
         CommCareApplication._().closeUserSession();
     }
 
+    public String getCurrentUserId() {
+        if (currentUserId != null) {
+            return currentUserId;
+        }
+        try {
+            currentUserId = this.getSession().getLoggedInUser().getUniqueId();
+            return currentUserId;
+        } catch (SessionUnavailableException e) {
+            return "";
+        }
+    }
+
+    private void refreshUserIdCache() {
+        try {
+            currentUserId = this.getSession().getLoggedInUser().getUniqueId();
+        } catch (SessionUnavailableException e) {

```

Comment:
Yea, I think I left that out b/c this only ever gets called right after we've bound the user session service, but should be there for good measure anyway.